src/modm/math/algorithm/prescaler.hpp

```diff
@@ -159,4 +161,6 @@ Prescaler
 	}
 };
 
+using Prescaler = GenericPrescaler<modm::frequency_t>;
+
```

Comment:
The intermediary results `uint32_t baud_lower` and `uint32_t baud_high` need to be `T` too, since a prescaler of 1 is possible and will overflow 32-bit.

Reply:
Also the cast in the calculation of `prescaler_floor` and `prescaler_ceiling` of `from_range()`

Reply:
Hm good point, if you have a *very* low desired frequency so that the prescaler >= 32-bit then there's probably a overflow there too…

We could just `T` it all.

Reply:
In fairness we already use `double`, which is not great for runtime, so making everything `uint64_t` isn't going to be an issue…